test/dlc_tests/sanity/quick_checks/test_yml_config.py

```diff
@@ -0,0 +1,57 @@
+import os
+import re
+
+import pytest
+import yaml
+
+from test.test_utils import is_pr_context
+
+
+@pytest.mark.quick_checks
+@pytest.mark.model("N/A")
+@pytest.mark.integration("release_images_training.yml")
+@pytest.mark.skipif(
+    not is_pr_context(),
+    reason="This test is only needed to validate release_images configs in PRs.",
+)
+def test_release_images_training_yml():
+    _release_images_yml_verifier(image_type="training", excluded_image_type="inference")
+
+
+@pytest.mark.quick_checks
+@pytest.mark.model("N/A")
+@pytest.mark.integration("release_images_inference.yml")
+@pytest.mark.skipif(
+    not is_pr_context(),
+    reason="This test is only needed to validate release_images configs in PRs.",
+)
+def test_release_images_inference_yml():
+    _release_images_yml_verifier(image_type="inference", excluded_image_type="training")
+
+
+def _release_images_yml_verifier(image_type, excluded_image_type):
+    """
+    Simple test to ensure release images yml file is loadable
+    Also test that excluded_image_type is not present in the release yml file
+    """
+    # Look up the path until deep-learning-containers is our base directory
+    dlc_base_dir = os.getcwd()
+    while os.path.basename(dlc_base_dir) != "deep-learning-containers":
+        dlc_base_dir = os.path.split(dlc_base_dir)[0]
+
+    release_images_yml_file = os.path.join(dlc_base_dir, f"release_images_{image_type}.yml")
+
+    # Define exclude regex
+    exclude_pattern = re.compile(rf"{excluded_image_type}", re.IGNORECASE)
+
+    with open(release_images_yml_file, "r") as release_imgs_handle:
+        for line in release_imgs_handle:
+            assert not exclude_pattern.search(
+                line
+            ), f"{exclude_pattern.pattern} found in {release_images_yml_file}. Please ensure there are not conflicting job types here."
```

Comment:
If we follow this strategy, the code will pick up the comment lines as well. If someone has mentioned a comment that has something like "Avoid train images", it would start failing. Is it possible to do `yaml.safe_load` first and then iterate through the loaded contents to see if it has a `train` keyword in it?

Reply:
+1 Shantanu.
Also is it easier just to add an if condition b/w L48 and L49?
if not line.lstrip().startswith('#'):

Reply:
Thanks all for your reviews. I think looking at yaml contents makes more sense than parsing out comment strings manually. I'll plan to make this change and re-test

Reply:
We will update test optimizations in subsequent PR to prioritize getting this functionality available to users